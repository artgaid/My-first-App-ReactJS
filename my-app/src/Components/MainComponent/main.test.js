import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Main from "./main";

describe("Main component", () => {
  it("Main render", () => {
    render(
      <Router>
        {" "}
        <Main />
      </Router>
    );

    expect(screen.getByText(/app/i)).toBeInTheDocument();
    expect(document.querySelector("a")).toBeInTheDocument();
  });

  it("Main snapshot", () => {
    const main = render(
      <Router>
        {" "}
        <Main />
      </Router>
    );

    expect(main).toMatchSnapshot();
  });
});
