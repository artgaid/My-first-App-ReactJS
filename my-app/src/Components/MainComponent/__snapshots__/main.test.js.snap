// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component Main snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
       
      <div
        class="MuiBox-root css-18615pc"
      >
        <p
          class="MuiTypography-root MuiTypography-button css-1f0on15-MuiTypography-root"
        >
          My App is on ReactJS
        </p>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-175bk0n-MuiGrid-root"
        >
          <a
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-fj1giv-MuiButtonBase-root-MuiButton-root"
            href="/chats"
            tabindex="0"
          >
            Chats
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <a
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-fj1giv-MuiButtonBase-root-MuiButton-root"
            href="/profile"
            tabindex="0"
          >
            Profile
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <a
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-fj1giv-MuiButtonBase-root-MuiButton-root"
            href="/pokemons"
            tabindex="0"
          >
            API Pokemons
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <a
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-fj1giv-MuiButtonBase-root-MuiButton-root"
            href="/pexels"
            tabindex="0"
          >
            API Pexels
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
     
    <div
      class="MuiBox-root css-18615pc"
    >
      <p
        class="MuiTypography-root MuiTypography-button css-1f0on15-MuiTypography-root"
      >
        My App is on ReactJS
      </p>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-175bk0n-MuiGrid-root"
      >
        <a
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-fj1giv-MuiButtonBase-root-MuiButton-root"
          href="/chats"
          tabindex="0"
        >
          Chats
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-fj1giv-MuiButtonBase-root-MuiButton-root"
          href="/profile"
          tabindex="0"
        >
          Profile
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-fj1giv-MuiButtonBase-root-MuiButton-root"
          href="/pokemons"
          tabindex="0"
        >
          API Pokemons
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-fj1giv-MuiButtonBase-root-MuiButton-root"
          href="/pexels"
          tabindex="0"
        >
          API Pexels
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
